ui/app/AppLayouts/Wallet/popups/SignTransactionModalBase.qml

```diff
@@ -66,12 +68,11 @@ StatusDialog {
     width: 480
     padding: 0
 
-    function formatBigNumber(number: string, decimals = -1) {
+    function formatBigNumber(number: string, symbol: string, options = null) {
         if (!number)
-            return ""
-        const big = SQUtils.AmountsArithmetic.fromString(number)
-        const resultNum = decimals === -1 ? big.toFixed() : big.round(decimals).toFixed()
-        return resultNum.replace('.', Qt.locale().decimalPoint)
+            return "N/A"
+        // NB: we're fine with some precision loss here for the purpose of displaying that number
+        return currencyStore.formatCurrencyAmount(parseFloat(number), symbol, options)
```

Comment:
Since now the `formatBigNumber` logic is just to call the store WDYT of removing it completely to remove the store dependency as Michal suggested? It's probably not worth wrapping the store functions in a base component.

I'm thinking for the popups can receive the formatted values instead.
We already do it for the fees in `SwapSignModal` and `SwapApproveCapModal`.
```
            fiatFees: root.swapAdaptor.currencyStore.formatCurrencyAmount(root.swapAdaptor.swapOutputData.totalFees,
                                                                          root.swapAdaptor.currencyStore.currentCurrency)
            cryptoFees: {
                const cryptoValue = root.swapAdaptor.currencyStore.getCryptoValue(root.swapAdaptor.swapOutputData.totalFees, Constants.ethToken)
                return root.swapAdaptor.currencyStore.formatCurrencyAmount(cryptoValue, Constants.ethToken)
            }
```

Reply:
Yeah, I think I'll just pass down the formatted values; otoh I still need the `currencyStore` for other purposes (existing code). I'll give it a look

Reply:
There are many options like taking formatted values. Or taking just a formatting function. I cannot find another usages of `currencyStore` in those popups. But if any, only specific popup could take it directly. Or ideally only what is needed. Making ui components not dependent on stores makes the api cleaner and will be beneficial in the future to better organize stores as well.

Reply:
Done